### PR TITLE
data-source/acl_policies: add new data source

### LINKED ISSUE #83: Create data source for listing ACL policies
# nomad_acl_policies

Get a list of [ACL policies](https://www.nomadproject.io/api/acl-policies.html#list-policies) filtered by prefix. If no prefix is provided all policies are returned.

## Example usage

```hcl
data "nomad_acl_policies" "stag" {
  prefix = "stag"
}
```

## Argument reference

* `prefix`: `(string)` Name prefix to filter policies.

## Attributes reference

* `policies`: `(list of maps)` A list of policies that match `prefix`.
    * `name`: `(string)` Name of the ACL policy.
    * `description`: `(string)` ACL policy description.

### PR DESCRIPTION
**Pull request by anGie44**
Closes #83 

Output of acceptance test:
```
--- PASS: TestAccDataSourceNomadAclPolicies_Basic (0.96s)
```

### CODE REVIEW

File: nomad/data_source_acl_policies_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+func TestAccDataSourceNomadAclPolicies_Basic(t *testing.T) {
+	dataSourceName := "data.nomad_acl_policies.test"
+	numPolicies := 2
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testProviders,
+		Steps: []resource.TestStep{
+			{
+				PreConfig:   testAccCreateNomadAclPolicies(t, numPolicies),
```
Comment (anGie44) [inline nomad/data_source_acl_policies_test.go:22]: work-around for creating acl policies; when done in a terraform config, the test resulted in a non-empty plan after refresh such that the individual ACL policy resources were getting deleted and the ACL policies data-source needed to update after refresh

File: nomad/data_source_acl_policies.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	log.Print(debugMsg)
+
+	policies, _, err := client.ACLPolicies().List(queryOpts)
+	if err != nil {
+		// As of Nomad 0.4.1, the API client returns an error for 404
+		// rather than a nil result, so we must check this way.
+		if strings.Contains(err.Error(), "404") {
+			d.SetId("")
+			return nil
+		}
```
Comment (cgbaker) [inline nomad/data_source_acl_policies.go:68]:
```suggestion
```

Comment (cgbaker) [reply nomad/data_source_acl_policies.go:68]: these 404 checks are more for querying specific objects (e.g., specific policies) by name. the policy list endpoint will not 404.

File: nomad/data_source_acl_policies.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		return fmt.Errorf("error getting ACL policies: %#v", err)
+	}
+
+	if len(policies) == 0 {
+		return errors.New("query returned an empty list of ACL policies")
+	}
+
```
Comment (cgbaker) [inline nomad/data_source_acl_policies.go:75]:
```suggestion
```

Comment (cgbaker) [reply nomad/data_source_acl_policies.go:75]: this seems like a valid, non-error case.

File: nomad/data_source_acl_policies.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	d.SetId(resource.UniqueId())
+	if err := d.Set("policies", flattenAclPolicies(policies)); err != nil {
+		return fmt.Errorf("error setting policies: %#v", err)
+	}
+
+	return nil
+}
+
+func flattenAclPolicies(policies []*api.ACLPolicyListStub) []interface{} {
+	output := make([]interface{}, 0)
```
Comment (cgbaker) [inline nomad/data_source_acl_policies.go:85]:
```suggestion
	output := make([]interface{}, 0, len(policies))
```

Comment (cgbaker) [reply nomad/data_source_acl_policies.go:85]: might as well initialize with the necessary capacity

File: nomad/provider.go [hunk 5]
```diff
@@ -65,12 +65,13 @@ func Provider() terraform.ResourceProvider {
...
 
 		DataSourcesMap: map[string]*schema.Resource{
-			"nomad_acl_policy":  dataSourceAclPolicy(),
-			"nomad_acl_token":   dataSourceACLToken(),
-			"nomad_deployments": dataSourceDeployments(),
-			"nomad_job":         dataSourceJob(),
-			"nomad_namespaces":  dataSourceNamespaces(),
-			"nomad_regions":     dataSourceRegions(),
+			"nomad_acl_policy":   dataSourceAclPolicy(),
+			"nomad_acl_policies": dataSourceAclPolicies(),
```
Comment (cgbaker) [inline nomad/provider.go:69]:
```suggestion
			"nomad_acl_policies": dataSourceAclPolicies(),
			"nomad_acl_policy":   dataSourceAclPolicy(),
```

Comment (cgbaker) [reply nomad/provider.go:69]: we strive to keep these alphabetical

File: nomad/data_source_acl_policies_test.go (hunk 1 above)
Comment (cgbaker) [reply nomad/data_source_acl_policies_test.go:22]: neat! i think i need to look at this for some of our other tricky testing scenarios.

**Review by lgfa29 — APPROVED** [review]

**Review by cgbaker — APPROVED** [review]
👍

**Pull request merged**